ersilia/utils/identifiers/compound.py

```diff
@@ -122,25 +125,23 @@ def chemical_identifier_resolver(identifier):
 
     def encode(self, smiles):
         """Get InChIKey of compound based on SMILES string"""
+        if not isinstance(smiles, str) or not smiles.strip() or smiles == UNPROCESSABLE_INPUT:
+            return UNPROCESSABLE_INPUT
+        
         if self.Chem is None:
-            inchikey = self._pubchem_smiles_to_inchikey(smiles)
-            if inchikey is None:
-                inchikey = self._nci_smiles_to_inchikey(smiles)
+            inchikey = self._pubchem_smiles_to_inchikey(smiles) or self._nci_smiles_to_inchikey(smiles)
         else:
             try:
                 mol = self.Chem.MolFromSmiles(smiles)
                 if mol is None:
-                    raise Exception(
-                        "The SMILES string: %s is not valid or could not be converted to an InChIKey"
-                        % smiles
-                    )
+                    return UNPROCESSABLE_INPUT
                 inchi = self.Chem.rdinchi.MolToInchi(mol)[0]
                 if inchi is None:
-                    raise Exception("Could not obtain InChI")
+                    return UNPROCESSABLE_INPUT
                 inchikey = self.Chem.rdinchi.InchiToInchiKey(inchi)
             except:
                 inchikey = None
-        return inchikey
+        
+        return inchikey if inchikey is not None else UNPROCESSABLE_INPUT
```

Comment:
This can simply be `return inchikey if inchikey else UNPROCESSABLE_INPUT`.  Remember the Zen of Python: Simple is better than complex ;)

Reply:
Thanks @DhanshreeA  updated